core/src/components/datetime/utils/data.ts

```diff
@@ -254,7 +254,22 @@ export const generateTime = (
  * Given DatetimeParts, generate the previous,
  * current, and and next months.
  */
-export const generateMonths = (refParts: DatetimeParts): DatetimeParts[] => {
+export const generateMonths = (refParts: DatetimeParts, forcedDate: DatetimeParts | null = null): DatetimeParts[] => {
+  /**
+   * If we're forcing a month to appear, and it's different from the current month,
+   * ensure it appears by replacing the next or previous month as appropriate.
+   */
+  if (forcedDate !== null && (refParts.month !== forcedDate.month || refParts.year !== forcedDate.year)) {
```

Comment:
Is there ever a time when `refParts` could be the same as `forcedDate`? In `processValue` we check for `didChangeMonth` which should cover this.

Reply:
Yes, `generateMonths` gets called again after the scroll to the forced date occurs but before `forceRenderDate` is reset due to Stencil's render cycle. Having the check in place here ensures that the forced month is only rendered as long as it needs to be.